pkg/commands/os_test.go

```diff
@@ -1,16 +1,299 @@
 package commands
 
-import "testing"
+import (
+	"os/exec"
+	"testing"
 
-func TestQuote(t *testing.T) {
-	osCommand := &OSCommand{
-		Log:      nil,
-		Platform: getPlatform(),
+	"github.com/stretchr/testify/assert"
+)
+
+func newDummyOSCommand() *OSCommand {
+	return NewOSCommand(newDummyLog())
+}
+
+func TestOSCommandRunCommandWithOutput(t *testing.T) {
+	type scenario struct {
+		command string
+		test    func(string, error)
 	}
-	test := "hello `test`"
-	expected := osCommand.Platform.escapedQuote + "hello \\`test\\`" + osCommand.Platform.escapedQuote
-	test = osCommand.Quote(test)
-	if test != expected {
-		t.Error("Expected " + expected + ", got " + test)
+
+	scenarios := []scenario{
+		{
+			"echo -n '123'",
+			func(output string, err error) {
+				assert.NoError(t, err)
+				assert.EqualValues(t, "123", output)
+			},
+		},
+		{
+			"rmdir unexisting-folder",
+			func(output string, err error) {
+				assert.Regexp(t, ".*No such file or directory.*", err.Error())
+			},
+		},
+	}
+
+	for _, s := range scenarios {
+		s.test(newDummyOSCommand().RunCommandWithOutput(s.command))
+	}
+}
+
+func TestOSCommandRunCommand(t *testing.T) {
+	type scenario struct {
+		command string
+		test    func(error)
+	}
+
+	scenarios := []scenario{
+		{
+			"rmdir unexisting-folder",
+			func(err error) {
+				assert.Regexp(t, ".*No such file or directory.*", err.Error())
+			},
+		},
 	}
+
+	for _, s := range scenarios {
+		s.test(newDummyOSCommand().RunCommand(s.command))
+	}
+}
+
+func TestOSCommandGetOpenCommand(t *testing.T) {
+	type scenario struct {
+		command func(string, ...string) *exec.Cmd
+		test    func(string, string, error)
+	}
+
+	scenarios := []scenario{
+		{
+			func(name string, arg ...string) *exec.Cmd {
+				return exec.Command("exit", "1")
+			},
+			func(name string, trail string, err error) {
+				assert.EqualError(t, err, "Unsure what command to use to open this file")
+			},
+		},
+		{
+			func(name string, arg ...string) *exec.Cmd {
+				assert.Equal(t, "which", name)
+				assert.Len(t, arg, 1)
+				assert.Regexp(t, "xdg-open|cygstart|open", arg[0])
```

Comment:
if we're returning a command that we know won't error, then the expected behaviour here is to return at xdg-open, meaning we don't need to check for cygstart or open (same with the test function below). 

It's also just occured to me that because there is some platform-related logic here that we should get rid of trailMap in GetOpenCommand() and attach those values to the *Platform struct, but that can be done in another PR.

Reply:
Here I do this because we have a map so it's non deterministic, anything could pop at each test cycle. 

I agree with your last comment.

Reply:
Ah that makes sense, I use a different language at work which would be deterministic in this case

Reply:
Moving config somewhere else would turn this test better.